lib/src/test/java/io/ably/lib/test/realtime/RealtimeAuthTest.java

```diff
@@ -933,6 +938,84 @@ public Object getTokenRequest(Auth.TokenParams params) {
         }
     }
 
+    @Test
+    public void auth_renewAuth_callback_invoked() throws InterruptedException {
+        try {
+            /* get a TokenDetails */
+            final String testKey = testVars.keys[0].keyStr;
+            final ClientOptions clientOptions = createOptions(testKey);
+            final AblyRest ablyRest = new AblyRest(clientOptions);
+
+            final TokenDetails tokenDetails = ablyRest.auth.requestToken(new Auth.TokenParams() {{
+                ttl = 1000L;
+            }}, null);
+            assertNotNull("Expected token value", tokenDetails.token);
+
+            // create Ably realtime instance with token and authCallback
+            class ProtocolListener extends DebugOptions implements DebugOptions.RawProtocolListener {
+                ProtocolListener() {
+                    Setup.getTestVars().fillInOptions(this);
+                    protocolListener = this;
+                }
+
+                @Override
+                public void onRawConnectRequested(String url) {
+                    synchronized (this) {
+                        notify();
+                    }
+                }
+
+                @Override
+                public void onRawConnect(String url) {
+                }
+
+                @Override
+                public void onRawMessageSend(ProtocolMessage message) {
+                }
+
+                @Override
+                public void onRawMessageRecv(ProtocolMessage message) {
+                }
+            }
+
+            final ProtocolListener protocolListener = new ProtocolListener();
+            protocolListener.autoConnect = false;
+            protocolListener.tokenDetails = tokenDetails;
+            //   implement callback, using Ably instance with key
+            protocolListener.authCallback = params -> tokenDetails;
+
+            final AblyRealtime ably = new AblyRealtime(protocolListener);
+            synchronized (protocolListener) {
```

Comment:
I was also surprised to see that. [Documentation on wait](https://docs.oracle.com/javase/7/docs/api/java/lang/Object.html#wait()) has an example that is wrapped in `synchronized` block while the same documentation says that it should be used in a loop which isn't in this case 🤔 Honestly I have never ever used `wait` directly in any code I wrote. If you think it will be worth it, I can replace this waits with a latch.

Reply:
Let's leave it for today, but I would appreciate it if you could find some time to take a look. My point of consideration is simply to study this `protocolListener` locally scoped variable and try and work out what competing code is being blocked by using it as a sync anchor. 😕